### ISSUE TITLE
How to disable `transparent ciphertext` runtime error?

### ISSUE THREAD

**Issue by AlexMV12** [ibarrond/Pyfhel#124]
**Describe the feature**
By default, SEAL enable a runtime check against "transparent" ciphertexts.
As of this [SEAL issue](https://github.com/microsoft/SEAL/issues/276), transparent ciphertexts are defined as ciphertexts whose second polynomial is `0`. 
The check is done because such ciphertext (in the form `(ctxt[0], 0)`) is just a "plain" value inside a ciphertext object, hence, not guaranteeing any protection against attacks (i.e., no secret key is needed to access that value). Producing a transparent ciphertext in Pyfhel results in a runtime error:

```python
from Pyfhel import Pyfhel, PyPtxt, PyCtxt

HE = Pyfhel()
HE.contextGen(scheme='bfv', n=2**12, t_bits=20)
HE.keyGen()
HE.relinKeyGen()

ctxt1 = HE.encrypt(42)
ptxt1 = HE.encode(0)

r = ctxt1*ptxt1
```
produces:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [119], in <cell line: 7>()
      4 ctxt1 = HE.encrypt(42)
      5 ptxt1 = HE.encode(0)
----> 7 r = ctxt1*ptxt1

File Pyfhel/PyCtxt.pyx:431, in Pyfhel.PyCtxt.PyCtxt.__mul__()

File Pyfhel/Pyfhel.pyx:1041, in Pyfhel.Pyfhel.Pyfhel.multiply_plain()

RuntimeError: result ciphertext is transparent
```

In the same [SEAL issue](https://github.com/microsoft/SEAL/issues/276) it is written that such warning may be disabled "[by] passing -DSEAL_THROW_ON_TRANSPARENT_CIPHERTEXT=OFF when configuring SEAL through CMake".

It would be nice to have a way to disable this check, what do you think? Thanks!

**Is this feature related to a project/objective? Please describe briefly**
Transparent ciphertexts may be expected in some scenarios. Consider, for instance, ML/DL models on encrypted data. It may happen that a weight is `0`; hence, it is not surprising that the output neuron is... `0`. It is not a problem, though, because no data leak happens.

Unfortunately a RuntimeError is quite difficult to manage, and for now I have no ideas on how to proceed.

**Comment by ibarrond** [comment]
I would leave it on by default for security purposes (so that novel users do not accidentally program vulnerable applications). That being said, you can control the CMake flags passed to SEAL by modifying the `pyproject.toml` file. In your case, you can manually set the following property to 'OFF' (and install the repo locally via `pip install .`) to disable that Runtime Error:

https://github.com/ibarrond/Pyfhel/blob/f55661978412b2cc6d4f6573d97f47f760afe85f/pyproject.toml#L102

**Comment by AlexMV12** [comment]
Thanks! This looks reasonable. Thank you for the commit.

**Comment by ibarrond** [comment]
Mmmmm, can you do `pip install -v .` (verbose) and check if there is any occurrence of `"SEAL_THROW_ON_TRANSPARENT_CIPHERTEXT"` in the output? This way we could rule out an issue with argument parsing to run CMake.

**Comment by AlexMV12** [comment]
Yes, there is, but the settings is not taken:

```
...
-- Performing Test C_FLAG_WCAST_ALIGN - Success
  -- Performing Test CXX_FLAG_WCAST_ALIGN
  -- Performing Test CXX_FLAG_WCAST_ALIGN - Success
  -- Performing Test C_FLAG_WCAST_QUAL
  -- Performing Test C_FLAG_WCAST_QUAL - Success
  -- Performing Test CXX_FLAG_WCAST_QUAL
  -- Performing Test CXX_FLAG_WCAST_QUAL - Success
  -- Performing Test C_FLAG_WSTRICT_PROTOTYPES
  -- Performing Test C_FLAG_WSTRICT_PROTOTYPES - Success
  -- CMAKE_INSTALL_PREFIX: /usr/local
  -- CMAKE_INSTALL_LIBDIR: lib
  -- ZSTD_LEGACY_SUPPORT not defined!
  -- ZSTD_MULTITHREAD_SUPPORT is disabled
  -- SEAL_USE_INTEL_HEXL: OFF
  -- BUILD_SHARED_LIBS: OFF
  -- SEAL_THROW_ON_TRANSPARENT_CIPHERTEXT: ON
  -- SEAL_USE_GAUSSIAN_NOISE: OFF
  -- SEAL_DEFAULT_PRNG: Blake2xb
  -- x86intrin.h - found
  -- SEAL_USE_INTRIN: ON
  -- Performing Test SEAL_MEMSET_S_FOUND
  -- Performing Test SEAL_MEMSET_S_FOUND - Failed
...
```

I confirm that I modified the pyproject file:
```
  ## CMake options: https://github.com/microsoft/SEAL/#basic-cmake-options
    [cpplibraries.SEAL.cmake_opts]
    CMAKE_BUILD_TYPE = 'Release'
    SEAL_USE_INTEL_HEXL ='OFF'   # ON/OFF, use Intel HEXL for low-level kernels
    SEAL_USE_ALIGNED_ALLOC ='OFF'# ON/OFF, 64B aligned memory allocations, better perf with Intel HEXL.
    BUILD_SHARED_LIBS ='OFF'     # ON/OFF, build shared and not static library
    SEAL_THROW_ON_TRANSPARENT_CIPHERTEXT = 'OFF' # ON/OFF, runtime error when multiplying a ctxt with a zeroed plaintext.
```

Thanks,
Alessandro

**Comment by ibarrond** [comment]
Should be wor

**Comment by AlexMV12** [comment]
Thanks, not it is working!
Let's try to close it then. :) Fixes #124
